desc/profiles.py

```diff
@@ -901,69 +1099,213 @@ def from_values(
-        order : int
-            polynomial order
-        xs : int or ndarray
-            x locations to use for fit. If an integer, uses that many points linearly
-            spaced between 0,1
-        rcond : float
-            Relative condition number of the fit. Singular values smaller than this
-            relative to the largest singular value will be ignored. The default value
-            is len(x)*eps, where eps is the relative precision of the float type, about
-            2e-16 in most cases.
-        w : array-like, shape(M,)
-            Weights to apply to the y-coordinates of the sample points. For gaussian
-            uncertainties, use 1/sigma (not 1/sigma**2).
+        params : array-like
+            polynomial coefficients to use, in ascending order. If not given, uses the
+            values given by the params attribute
+        grid : Grid or array-like
+            locations to compute values at. Defaults to self.grid
+        dr, dt, dz : int
+            derivative order in rho, theta, zeta
 
         Returns
         -------
-        profile : PowerSeriesProfile
-            profile in power series form.
+        values : ndarray
+            values of the profile or its derivative at the points specified
 
         """
-        if np.isscalar(xs):
-            xs = np.linspace(0, 1, xs)
-        fs = self.compute(grid=xs)
-        p = PowerSeriesProfile.from_values(xs, fs, order, rcond=rcond, w=w)
-        p.grid = self.grid
-        p.name = self.name
-        return p
+        if params is None:
+            params = self.params
+        transform = self._get_transform(grid)
+        return transform.transform(params, dr=dr, dt=dt, dz=dz)
 
-    def to_spline(self, knots=20, method="cubic2"):
-        """Convert this profile to a SplineProfile
+    @classmethod
+    def from_values(cls, r, t, z, f, L=6, M=0, N=0, NFP=1, w=None, grid=None, name=""):
+        """Fit a FourierZernikeProfile from point data.
 
         Parameters
         ----------
-        knots : int or ndarray
-            x locations to use for spline. If an integer, uses that many points linearly
-            spaced between 0,1
-        method : str
-            method of interpolation
-            - `'nearest'`: nearest neighbor interpolation
-            - `'linear'`: linear interpolation
-            - `'cubic'`: C1 cubic splines (aka local splines)
-            - `'cubic2'`: C2 cubic splines (aka natural splines)
-            - `'catmull-rom'`: C1 cubic centripedal "tension" splines
+        r, t, z : array-like, shape(k,)
+            coordinate locations in rho, theta, zeta
+        f : array-like, shape(k,)
+            function values
+        L, M, N : int
+            maximum mode numbers to fit
+        NFP : int
+            number of field periods
+        w : array-like, shape(k,)
+            Weights to apply to the y-coordinates of the sample points. For gaussian
+            uncertainties, use 1/sigma (not 1/sigma**2).
```

Comment:
why this note on 1/sigma**2?

Reply:
if its fitted from experimental data, this is the same convention as ``np.polyfit``